src/main/java/org/folio/services/finance/transaction/BaseTransactionService.java

```diff
@@ -90,20 +90,26 @@ public Future<Void> updateTransaction(Transaction transaction, RequestContext re
   }
 
   public Future<Void> updateTransactions(List<Transaction> transactions, RequestContext requestContext) {
-    var semaphore = new Semaphore(5, requestContext.getContext().owner());
-    List<Future<Void>> futures = new ArrayList<>();
-
+    if (transactions.isEmpty()) {
+      return Future.succeededFuture();
+    }
     return requestContext.getContext()
-      .executeBlocking(promise -> {
+      .<List<Future<Void>>>executeBlocking(promise -> {
+        List<Future<Void>> futures = new ArrayList<>();
+        var semaphore = new Semaphore(1, requestContext.getContext().owner());
```

Comment:
Do we use 1 active thread to bypass issues with concurrent access when working with transactions until AllOrNothing refactoring will be completed? I think it makes sense, thanks

Reply:
exactly